Nim/Nim/Nim.cpp

```diff
@@ -0,0 +1,133 @@
+using namespace std;
+
+#include <iostream>
+#include <string>
+#include <cmath>
+
+int MatchSelect(int Num, int Total, string PlayerName) // I always get the total matches left for other player from here.
+{
+    if (Num > 0 && Num < 4)
+    {
+        Total -= Num;
+    }
+    else
+    {
+        cout << "Invalid Number, you missed your chance to decide how many matches to remove from the table" << endl;
+        Num = rand() % 3 + 1;
+        Total -= Num;
+        cout << "Computer automaticly removed " << Num << " matches from the table for your turn." << endl;
+    }
+    if (Total == 0)
+    {
+        cout << "You win " << PlayerName << "' you are the winner!" << endl;
+    }
+    return Total;
+}
+
+bool GameFinished(int Total)
+{
+    if (Total >= 0) return false;
```

Comment:
`return total <= 0;` -- also, there is a bug here, because the game does not finish, if someone took the last straw, but only on the turn after that.